src/main/java/io/lighty/yang/validator/formats/JsTree.java

```diff
@@ -162,46 +175,55 @@ private void printLines(final List<Line> lines) {
             }
         }
         for (DataSchemaNode node : module.getChildNodes()) {
+            schemaInferenceStack.enterSchemaTree(node.getQName());
             final ArrayList<Integer> ids = singletonListInitializer.getSingletonListWithIncreasedValue();
             LyvNodeData lyvNodeData = new LyvNodeData(this.schemaContext, node, Collections.emptyList());
             HtmlLine htmlLine = new HtmlLine(ids, lyvNodeData, RpcInputOutput.OTHER, removeChoiceQnames,
                     namespacePrefix);
             lines.add(htmlLine);
             resolveChildNodes(lines, new ArrayList<>(ids), node, RpcInputOutput.OTHER, removeChoiceQnames,
-                    Collections.emptyList());
+                    Collections.emptyList(), schemaInferenceStack);
+            schemaInferenceStack.exit();
         }
         return lines;
     }
 
     private List<Line> getAugmentationNodesLines(final ArrayList<Integer> ids, final AugmentationSchemaNode augNode,
```

Comment:
You can change final ArrayList<Integer> ids to final List<Integer> ids here, too.

Reply:
Done